### PR TITLE
Add key type tests

### PR DESCRIPTION
**Pull request by aljones15**
Refactors the test suite so a common set of assertions runs on all the curves and not just one of them.

Addresses: https://github.com/digitalbazaar/ecdsa-multikey/issues/34

### CODE REVIEW

File: test/mock-data.js [hunk 1]
```diff
@@ -9,22 +9,55 @@ export const mockKey = {
 };
 
 export const mockKeyEcdsaSecp256 = {
-  type: 'EcdsaSecp256r1VerificationKey2019',
+  type: 'Multikey',
```
Comment (dlongley) [inline test/mock-data.js:12]: This one needs to be reverted back to `EcdsaSecp256r1VerificationKey2019`. This is for backwards compatibility tests. The others should be `Multikey` as they are now.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: test/mock-data.js (hunk 1 above)
Comment (aljones15) [inline test/mock-data.js:12]:
```suggestion
  type: 'EcdsaSecp256r1VerificationKey2019',
```

Comment (aljones15) [reply test/mock-data.js]: Reverted here: https://github.com/digitalbazaar/ecdsa-multikey/pull/35/commits/28c76f394ed4919f39581d2fdc60611810a2dd92 but that breaks the id should be set test so will need to further revise this PR.

Comment (aljones15) [reply test/mock-data.js]: @dlongley ok so inside of the `multkeys` map in mock-data I now set the type to `Multikey` so the P-256 one will pass the should set id test: https://github.com/digitalbazaar/ecdsa-multikey/pull/35/files/28c76f394ed4919f39581d2fdc60611810a2dd92..09a57e00aeebfdc1d25a6e6eee4c64273e64a5ff

Should we test both the backwards compatible `EcdsaSecp256r1VerificationKey2019` P-256 and the `Multikey` `P-256`? if so I'm assuming for `EcdsaSecp256r1VerificationKey2019` the id is not set by default?

This is the test that is failing https://github.com/digitalbazaar/ecdsa-multikey/blob/09a57e00aeebfdc1d25a6e6eee4c64273e64a5ff/test/assertions.js#L173-L178 for `EcdsaSecp256r1VerificationKey2019`

Comment (dlongley) [reply test/mock-data.js]: @aljones15, we shouldn't hack the key type to be `Multikey` just to pass this test. Either the library is supposed to auto-generate an `id` no matter what the key type was, or it shouldn't do it for a key type of `EcdsaSecp256r1VerificationKey2019`. For now, I'd say to skip this test if the key type isn't `Multikey` and file an issue to possibly add `id` auto-generation for other key types.

Comment (aljones15) [reply test/mock-data.js]: @dlongley not sure I hacked the key type to be Multikey as by default `generate({curve: 'P-256'})` produces a key with type `Multikey`. All the test suite does is ensure that all keys in the `multikeys` mock-data map have type `Multikey`. The original `EcdsaSecp256r1VerificationKey2019` is not mutated and still exists in the suite. The question is should we test both P-256 Multikey and P-256 EcdsaSecp256r1VerificationKey2019, the answer to that appears to be file an issue about it for now as `EcdsaSecp256r1VerificationKey2019` does not pass all the tests.

Comment (aljones15) [reply test/mock-data.js]: issue created here: https://github.com/digitalbazaar/ecdsa-multikey/issues/36

Comment (aljones15) [reply test/mock-data.js]: @dlongley the original key type import tests were left as is: https://github.com/digitalbazaar/ecdsa-multikey/blob/09a57e00aeebfdc1d25a6e6eee4c64273e64a5ff/test/EcdsaMultikey.spec.js#L66-L77

Comment (dlongley) [reply test/mock-data.js]: Ok, sounds good.

**Review by dlongley — APPROVED** [review]
I'm not sure about this style, but I don't have a better suggestion for reuse that is as succinct, so I'm going to approve and merge this. It's hard to tell that everything was preserved so I'm just assuming it is -- and maybe some codecov printout will tell us that things still get the same or better coverage somewhere.

**Comment by dlongley** [comment]
Thanks! Going to merge this now.

**Pull request merged**